packages/components/src/components/radio-group/radio-group.tsx

```diff
@@ -42,15 +42,21 @@ const RadioGroup = ({
         onToggle(e);
     };
 
+    const handleClick = (selected_option: string) => {
+        setSelectedOption(selected_option);
+    };
+
     return (
         <div className={classNames('dc-radio-group', className)}>
             {Array.isArray(children) &&
                 children.map(item => (
                     <ItemWrapper key={item.props.value} should_wrap_items={should_wrap_items}>
                         <label
+                            htmlFor={item.props.id}
                             className={classNames('dc-radio-group__item', className, {
                                 'dc-radio-group__item--selected': selected_option === item.props.value,
                             })}
+                            onClick={() => handleClick(item.props.value)}
```

Comment:
Why do we have **onClick** on `label` when the same action is handled in `input` ?

Reply:
+1 now we're the same handler on two different places for the same element

Reply:
the idea was to give a click handler for the label at the lowest level/the final touchpoint(which is this shared component). anyway i have rolled back this changes as matin suggested. 🏃🏼